packages/teleport/src/Nodes/AddNode/Iam/Iam.tsx

```diff
@@ -0,0 +1,267 @@
+            {attempt.status === 'failed' && (
+              <Alert kind="danger" children={attempt.statusText} />
+            )}
+            <Text mb={4}>
+              {`Add any Teleport Node or Proxy
+              with access to IAM credentials. `}
+              {`You can read more
+              about the IAM method in the `}
+              <Link href="https://goteleport.com/docs/setup/guides/joining-nodes-aws/">
+                documentation
+              </Link>
+              .
+            </Text>
+            <Box mb={4}>
+              <Text bold as="span">
+                Step 1
+              </Text>{' '}
+              - Download Teleport package to your computer
+              <DownloadLinks isEnterprise={isEnterprise} version={version} />
+            </Box>
+            <Box>
+              <Text bold as="span">
+                Step 2
+              </Text>{' '}
+              - Setup AWS Rules
+              {rules.map((rule, index) => (
+                <RuleBox
+                  key={index}
+                  bg="primary.lighter"
+                  borderRadius="2"
+                  mt={1}
+                  p="3"
+                  position="relative"
+                >
+                  <Text typography="h5">{`Rule #${index + 1}`}</Text>
+                  {rule.isCollapsed && (
+                    <ButtonEditRule
+                      title="edit rule"
+                      onClick={() =>
+                        setRuleAtIndex(index, { ...rule, isCollapsed: false })
+                      }
+                    >
+                      <Pencil />
+                    </ButtonEditRule>
+                  )}
+                  {!rule.isCollapsed && (
+                    <>
+                      {index !== 0 && (
+                        <ButtonRemoveRule onClick={() => handleRemove(index)}>
+                          Remove
+                        </ButtonRemoveRule>
+                      )}
+                      <Box>
+                        <FieldInput
+                          label="AWS Account"
+                          autoFocus
+                          onChange={e =>
+                            setRuleAtIndex(index, {
+                              ...rules[index],
+                              awsAccount: e.target.value,
+                            })
+                          }
+                          rule={value => requiredAwsAccount(value, rule)}
+                          placeholder="111111111111"
+                          value={rule.awsAccount}
+                        />
+                      </Box>
+                      <FieldInput
+                        mb={2}
+                        label="AWS ARN"
+                        onChange={e =>
+                          setRuleAtIndex(index, {
+                            ...rules[index],
+                            awsArn: e.target.value,
+                          })
+                        }
```

Comment:
when i see `setXXX` i think of hook :thinking: , how about `updateRule` instead?
also, i'd make the code a bit leaner by moving the logic to the helper func so like `updateRule(index, e?)`

Reply:
(marking as resolved since this is not a problem with a single rule)